h2o-core/src/main/java/water/rapids/PermutationVarImp.java

```diff
@@ -0,0 +1,310 @@
+package water.rapids;
+
+import hex.*;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.ArrayUtils;
+import water.util.TwoDimTable;
+import water.util.VecUtils;
+
+import java.util.*;
+
+/**
+ * Permutation Variable (feature) importance measures the increase in the prediction error of the model after permuting 
+ * the variables's values, which breaks the relationship between the variables and the true outcome.
+ * https://christophm.github.io/interpretable-ml-book/feature-importance.html
+ *
+ * Calculate permutation variables importance, by shuffling randomly each variables of the training Frame,
+ * scoring the model with the newly created frame using One At a Time approach
+ * and Morris method; creating TwoDimTable with relative, scaled, and percentage value
+ *                             TwoDimTable with mean of the absolute value, and standard deviation of all features importance
+ */
+
+public class PermutationVarImp {
+
+    public double[] _pVarImp; // permutation variables importance, (relative)
+    
+    private String _responseCol; 
+    
+    private String[] _variables; 
+    private String[] _varsToShuffle;
+    public String[] _metrics;
+
+    /**
+     * Stores the metric to be used (loss function) original values and j-th variables's score metric values  
+     */
+    public class LocalMetric{ // do it as a string to get the metric from the user
+
+        LocalMetric(String metric)  { _metric = metric; }
+        LocalMetric()               { _metric = "mse"; } 
+        public String _metric;
+        double _ogMetric;
+        double _variablesMetric;
+    }
+
+    public LocalMetric _varImpMetric;
+    public Map<String, Double> _varImpMap;
+    TwoDimTable _permutationVarImp;
+    private Model _model;
+    private Frame _inputFrame;
+
+    /**
+     * Constructor that stores the model, frame, response column, variable Strings and 
+     * sets the allowed metrics
+     * @param model trained model
+     * @param fr training frame
+     */
+    public PermutationVarImp(Model model, Frame fr) {
+        _model = model;
+        _inputFrame = fr;
+        _responseCol = _model._parms._response_column;
+        _variables = _inputFrame.names();
+        _metrics = new String[]{"r2", "mse", "rmse", "gini coefficient", "f1", "logloss", "auc"};
+    }
+    
+    /**
+     * Creates a new array of Strings without the response column and ignored columns
+     */
+    public void removeResCol(){
```

Comment:
You should rename this function if you are removing not only the response column.